test/gui/shared/steps/server_context.py

```diff
@@ -207,3 +207,28 @@ def step(context, user):
 )
 def step(context, user, resource, receiver, permissions):
     sharing_helper.share_resource(user, resource, receiver, permissions)
+
+
+@Given(
+    'user "|any|" has created a public link in the server with the following settings'
+)
+def step(context, user):
+    data = {
+        'resource': '',
+        'name': '',
+        'password': '',
+        'expireDate': '',
+        'permissions': 'read',
```

Comment:
```suggestion
        'resource': '',
        'permissions': '',
```

Reply:
i think for `permissions` we should keep default value as `read`

Reply:
but not internally in the step def. we need to provide permissions from the feature file

Reply:
okay i will be adding `permissions` as `read` (default) in the data table